File: Objective-C/LearnSwiftLanguage/LearnSwiftLanguage/BookViewController.m
//
//  ViewController.m
//  LearnSwiftLanguage
//
//  Created by bob on 16/11/9.
//  Copyright © 2016年 __company__. All rights reserved.
//

#import "BookViewController.h"
#import "BookTitleCell.h"
#import <SafariServices/SafariServices.h>
#import "YDHTTPSessionManager.h"

@interface BookViewController ()
@property (strong, nonatomic) NSMutableArray *dataSource;
@end

@implementation BookViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
    self.title = @"swift基础语法";
    [self.tableView registerNib:[UINib nibWithNibName:@"BookTitleCell" bundle:nil] forCellReuseIdentifier:@"BookTitleCell"];
    
    NSData *JSONData = [NSData dataWithContentsOfFile:[[NSBundle mainBundle] pathForResource:@"data" ofType:@"json"]];
    
    NSDictionary *data = [NSJSONSerialization JSONObjectWithData:JSONData options:NSJSONReadingAllowFragments error:nil];
    
    self.dataSource = data[@"swift"];
    
//    [[YDHTTPSessionManager sharedManager]requestDataWithPath:@"https://www.cnswift.org/" andBlock:^(id data, NSError *error) {
//        NSString *retStr = [[NSString alloc] initWithData:data encoding:NSUTF8StringEncoding];
//        NSLog(@"html = %@",retStr);
//    }];
    
}

- (NSInteger)numberOfSectionsInTableView:(UITableView *)tableView
{
    return 1;
}

- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section
{
    return self.dataSource.count;
}

- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath
{
    BookTitleCell *cell = [tableView dequeueReusableCellWithIdentifier:@"BookTitleCell" forIndexPath:indexPath];
    cell.name.text = self.dataSource[indexPath.row][@"name"];
    return cell;
}

- (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath
{
    [tableView deselectRowAtIndexPath:indexPath animated:YES];
    
    NSURL *url = [NSURL URLWithString:self.dataSource[indexPath.row][@"url"]];
    
    [self presentViewController:[[SFSafariViewController alloc]initWithURL:url] animated:YES completion:NULL];
    
}

@end

File: Swift/LearnSwiftLanguage/LearnSwiftLanguage/BookViewController.swift
//
//  BookViewController.swift
//  LearnSwiftLanguage
//
//  Created by bob on 16/12/12.
//  Copyright © 2016年 bob. All rights reserved.
//

import UIKit
import Foundation
import SafariServices

class BookViewController: UITableViewController {

    var dataSource: [[String : AnyObject]] = []
    
    override func viewDidLoad() {
        super.viewDidLoad()

        self.title = "swift基础语法";

        let file = Bundle.main.path(forResource: "data", ofType: "json") // Optional String
        let url = URL(fileURLWithPath: file!) //拆包
        
        // 一
//        do {
//            let data = try Data(contentsOf: url)
//            let allBooks = try JSONSerialization.jsonObject(with: data, options: JSONSerialization.ReadingOptions.allowFragments) as! [String : AnyObject]
//            if let arrJSON = allBooks["swift"] {
//                dataSource = arrJSON as! [[String : AnyObject]]
//            }
//        }
//        catch {
//            
//        }
        
        // 二
        
        let data = try? Data(contentsOf: url) //Optional Data
        let allBooks = try? JSONSerialization.jsonObject(with: data!, options: .allowFragments) as! [String : AnyObject] ////Optional [String : AnyObject]
        
        if let arrJSON = allBooks?["swift"] {
            dataSource = arrJSON as! [[String : AnyObject]]
        }
        
    }

    override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        return self.dataSource.count
    }


    override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
        let cell = tableView.dequeueReusableCell(withIdentifier: "BookTitleCell", for: indexPath) as! BookTitleCell
        cell.bookNameLabel.text = self.dataSource[indexPath.row]["name"] as? String

        return cell
    }
    
    override func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
        
        let bookurl = self.dataSource[indexPath.row]["url"] as! String
        
        present(SFSafariViewController.init(url: URL.init(string: bookurl)!), animated: true, completion: nil)
    }

}

